src/pages/filter/common/attributes/base-functions.store.ts

```diff
@@ -0,0 +1,140 @@
+import { makeAutoObservable, runInAction } from 'mobx'
+
+import { DefaultProblemGroup } from '@core/enum/default-problem-group-enum'
+import { FilterKindEnum } from '@core/enum/filter-kind.enum'
+import { ModeTypes } from '@core/enum/mode-types-enum'
+import datasetStore from '@store/dataset/dataset'
+import dtreeStore from '@store/dtree'
+import filterStore from '@store/filter'
+import activeStepStore from '@pages/filter/dtree/components/active-step.store'
+import {
+  IFuncPropertyStatus,
+  IInheritanceModeArgs,
+  TCondition,
+  TFuncCondition,
+  TPropertyStatus,
+  TVariant,
+} from '@service-providers/common'
+import { IStatfuncArguments } from '@service-providers/filtering-regime'
+import filteringRegimeProvider from '@service-providers/filtering-regime/filtering-regime.provider'
+import { getConditionJoinMode } from '@utils/getConditionJoinMode'
+import { getCurrentModeType } from '@utils/getCurrentModeType'
+interface IBaseAttributeStoreParams {
+  getAttributeStatus: () => TPropertyStatus | undefined
+  getInitialCondition: () => TCondition | undefined
+}
+
+export class BaseFunctionsStore {
```

Comment:
You duplicate some of code from BaseAttributeStore.
I think it will be better to extend BaseAttributeStore class or provide instance of BaseAttributeStore as dependency